dropwizard-jersey/src/main/java/io/dropwizard/jersey/validation/FuzzyEnumParamConverterProvider.java

```diff
@@ -0,0 +1,69 @@
+package io.dropwizard.jersey.validation;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Strings;
+import io.dropwizard.jersey.errors.ErrorMessage;
+import io.dropwizard.util.Enums;
+
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ParamConverter;
+import javax.ws.rs.ext.ParamConverterProvider;
+import javax.ws.rs.ext.Provider;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+
+import static io.dropwizard.jersey.validation.JerseyParameterNameProvider.getParameterNameFromAnnotations;
+
+/**
+ * Provides converters to jersey for enums used as resource parameters.
+ *
+ * <p>By default jersey will return a 404 if a resource parameter of an enum type cannot be converted. This class
+ * provides converters for all enum types used as resource parameters that provide better error handling. If an
+ * invalid value is provided for the parameter a {@code 400 Bad Request} is returned and the error message will
+ * include the parameter name and a list of valid values.</p>
+ */
+@SuppressWarnings("unchecked")
+@Provider
+public class FuzzyEnumParamConverterProvider implements ParamConverterProvider {
+    @Override
+    public <T> ParamConverter<T> getConverter(Class<T> rawType, Type genericType, Annotation[] annotations) {
+        if (!rawType.isEnum()) {
+            return null;
+        }
+
+        final Class<Enum<?>> type = (Class<Enum<?>>) rawType;
+        final Enum<?>[] constants = type.getEnumConstants();
+        final String parameterName = getParameterNameFromAnnotations(annotations).orElse("Parameter");
+
+        return new ParamConverter<T>() {
+            @Override
+            public T fromString(String value) {
+                if (Strings.isNullOrEmpty(value)) {
+                    return null;
+                }
+
+                Enum<?> constant = Enums.fromStringFuzzy(value, constants);
+                if (constant != null) {
+                    return (T) constant;
+                }
+
+                throw new WebApplicationException(getErrorResponse(String.format("%s must be one of [%s]", parameterName, Joiner.on(", ").join(constants))));
```

Comment:
This line's pretty long. Let's break it up and move the `Joiner` to a static field:

```
final String errorMsg = String.format(
    "%s must be one of [%s]", parameterName, joiner.join(constants));
throw new WebApplicationException(getErrorResponse(errorMsg));
```